### PR TITLE
New package: dnsdist-1.6.0

### PR DESCRIPTION
**Pull request by jailbird777**
#### General
- [X] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [X] I built this PR locally for my native architecture, (x86-64-musl)
- [X] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [X] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

This requires https://github.com/void-linux/void-packages/pull/28067

### CODE REVIEW

**Comment by ericonr** [comment]
What is the plan here? Is tinycdb useful if this isn't merged? If not much, this should all probably be a single PR.

**Comment by jailbird777** [comment]
Good question! Other distros separate them out since postfix & powerDNS at least can both use tinycdb also, so that it's shared.  However, if there's no plans to add CDB support to postfix or to add pdns (I don't personally have use for pdns), I can drop tiyncdb completely and just use the built-in CDB code in dnsdist instead.  I'm fine with either one!

**Comment by ericonr** [comment]
Guess I'll ask the postfix maintainer. @benalb do you have any interest in adding cdb support to postfix?

**Comment by benalb** [comment]
I do not use cdb myself, so I will not be able to test it. On the other hand, [this](http://www.postfix.org/CDB_README.html) seems pretty straightforward. I don't see why not.

**Comment by ericonr** [comment]
Ah, I had forgotten this. Maybe using bundled tinycdb instead of #28067 is better.

**Comment by jailbird777** [comment]
Removed the tinycdb dependency

**Comment by jailbird777** [comment]
I have an entire site that has been using this constantly since I first packaged it without any issues (at least since I switched from LuaJIT to Lua).  Is there a way to get this finally merged?

File: srcpkgs/dnsdist/template [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ --with-libcrypto=${XBPS_CROSS_BASE}/usr --with-service-user=_dnsdist
+ --with-service-group=_dnsdist --sbindir=/usr/bin --with-gnutls --with-lmdb
+ --enable-dns-over-tls --enable-dns-over-https --sysconfdir=/etc/dnsdist"
+conf_files="/etc/dnsdist/dnsdist.conf"
+hostmakedepends="autoconf automake pkgconf"
+makedepends="libsodium-devel boost-devel re2-devel net-snmp-devel libcap-devel
+ lua54-devel openssl-devel lmdb-devel fstrm-devel libedit-devel gnutls-devel
+ h2o-devel"
+depends="libsodium re2 libnet-snmp libcap lua54 openssl lmdb fstrm libedit
+ gnutls"
```
Comment (ericonr) [inline srcpkgs/dnsdist/template:17]: Most of these are detected automatically by xbps-src:

```
 SONAME: liblua5.4.so.5.4 <-> lua54>=5.4.0_1
   SONAME: libedit.so.0 <-> libedit>=20130712.3.1_1
   SONAME: libsodium.so.23 <-> libsodium>=1.0.15_1
   SONAME: libfstrm.so.0 <-> fstrm>=0.5.0_1
   SONAME: libnetsnmpmibs.so.40 <-> libnet-snmp>=5.9_2
   SONAME: libnetsnmpagent.so.40 <-> libnet-snmp>=5.9_2
   SONAME: libnetsnmp.so.40 <-> libnet-snmp>=5.9_2
   SONAME: libcap.so.2 <-> libcap>=2.16_1
   SONAME: libre2.so.7 <-> re2>=2020.06.01_1
   SONAME: libssl.so.1.1 <-> libssl1.1>=1.1.1f_1
   SONAME: libcrypto.so.1.1 <-> libcrypto1.1>=1.1.1f_1
   SONAME: liblmdb.so.0 <-> lmdb>=0.9.18_2
   SONAME: libgnutls.so.30 <-> gnutls>=3.4.13_1
   SONAME: libpthread.so.0 <-> glibc>=2.32_1
   SONAME: libstdc++.so.6 <-> libstdc++>=4.4.0_1
   SONAME: libm.so.6 <-> glibc>=2.32_1
   SONAME: libgcc_s.so.1 <-> libgcc>=4.4.0_1
   SONAME: libc.so.6 <-> glibc>=2.32_1
```

File: srcpkgs/dnsdist/template [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+homepage="https://dnsdist.org/"
+distfiles="https://downloads.powerdns.com/releases/${pkgname}-${version}.tar.bz2"
+checksum=a7783a04d8d4ad2b0168ffaaf85ef95d5f557057b0462280684dd799d0cdd292
+
+system_accounts="_dnsdist"
+
+pre_configure() {
+	export CPPFLAGS="-I${XBPS_CROSS_BASE}/usr/include"
+	export LDFLAGS="-L${XBPS_CROSS_BASE}/usr/lib"
+	export PKG_CONFIG_PATH=${XBPS_CROSS_BASE}/usr/lib/pkgconfig
```
Comment (ericonr) [inline srcpkgs/dnsdist/template:30]: These three shouldn't be necessary, have you tried building without them?

File: srcpkgs/dnsdist/template [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	export LMDB_LIBS="-llmdb"
+}
+
+post_install() {
+	vsconf dnsdistconf.lua dnsdist.conf
+	vinstall ${FILESDIR}/dnsdist.conf 644 etc/dnsdist
```
Comment (ericonr) [inline srcpkgs/dnsdist/template:36]: This should probably use the `dnsdist.conf` upstream provides to install under `/etc`? Why did you go with a custom file?

**Review by ericonr — COMMENTED** [review]
Please look at check results, there are still failures. You should have a single `New package:...` commit, not multiple ones.

And 32-bit builds are broken because they require y2038 support, apparently. You can add a check for `XBPS_TARGET_WORDSIZE = 32` and set `broken="no y2038 support"`, with a comment that this can be fixed for musl when we update it. Patching https://github.com/PowerDNS/pdns/blob/cec701c1f2881059c5bd1359d3720c555c7c1a3d/m4/pdns_check_time_t.m4 would require a bit more fiddling.

**Comment by jailbird777** [comment]
I've somehow stupidly messed up my git repo beyond recognition, so I'm closing this PR and opening a new one, sorry.

**Pull request closed without merging**